### PR TITLE
Check inherited interfaces exist

### PR DESCRIPTION
**Pull request by lukebjerring**
Reattempt at https://github.com/w3c/web-platform-tests/pull/10240, accommodating for dictionaries.

Chromium equivalent CL:
https://chromium-review.googlesource.com/#/c/chromium/src/+/992880
Note: Chromium drops the unit tests for idlharness.js, so IMO the CL isn't useful other than to see affected tests.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/10285)

### CODE REVIEW

**Comment by inexorabletash** [comment]
What about @foolip's suggestion on the earlier iteration of this to try landing this fix via Chrome's CQ instead, so we can get CI about any failures and update test expectations ahead of time?

**Comment by lukebjerring** [comment]
Doing that - just needed one of the 2 URLs to exist first :)

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-04-03 17:06:38
Finished: 2018-04-03 17:07:46





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/26778)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/361732283)

File: resources/idlharness.js [hunk 1]
```diff
@@ -682,6 +682,20 @@ IdlArray.prototype.test = function()
     }
     this["includes"] = {};
 
+    // Assert B defined for A : B
+    for (var member of Object.values(this.members).filter(m => m.base)) {
+        let lhs = member.name;
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply resources/idlharness.js]: Done.

File: resources/idlharness.js [hunk 2]
```diff
@@ -682,6 +682,20 @@ IdlArray.prototype.test = function()
     }
     this["includes"] = {};
 
+    // Assert B defined for A : B
+    for (var member of Object.values(this.members).filter(m => m.base)) {
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply resources/idlharness.js]: Done.

File: resources/idlharness.js [hunk 3]
```diff
@@ -682,6 +682,20 @@ IdlArray.prototype.test = function()
...
+        let lhs = member.name;
+        let rhs = member.base;
+        if (!(lhs in this.members)) throw new IdlHarnessError(`${lhs} inherits ${rhs}, but ${lhs} is undefined.`);
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply resources/idlharness.js]: Copy-pasta from code-block above (for which lhs could, in theory, fail to exist). We're iterating values of `this.members` here though, so `lhs in this.members` has to always be true. Removed.

**Comment by lukebjerring** [comment]
https://test-results.appspot.com/data/layout_results/linux_chromium_rel_ng/60373/layout-test-results/results.html

    Regressions: 34

**Comment by foolip** [comment]
Looks like most of the affected tests are due to base dictionaries not existing. Per https://github.com/w3c/web-platform-tests/issues/7146 dictionaries aren't really tested in any meaningful way, so an option could be to only do interfaces as a first step. But that'd leave us with a curious inconsistency, so either way is fine with me.

**Review by foolip — APPROVED** [review]
Changes LGTM, approving so this can be landed when the affected tests are updated.

**Comment by lukebjerring** [comment]
https://test-results.appspot.com/data/layout_results/linux_chromium_rel_ng/62438/layout-test-results/results.html

We're now down to 17 'regressions' which change or add passing subtests.

**Comment by lukebjerring** [comment]
Build failure is a stability check timeout.

**Comment by Hexcles** [comment]
Since idlharness.js is modified, all IDL tests are affected. Stability checks are likely to timeout in such case (they need to run the tests 10 times).

Interestingly though, Chrome stability check didn't time out and the results were stable. This, along with the Chromium CI results, should give us enough confidence. @foolip may admin-merge this PR once he see fit.

**Pull request merged**